src/js/components/SelectMultiple/SelectMultiple.js

```diff
@@ -434,10 +456,22 @@ const SelectMultiple = forwardRef(
                           !value || value?.length === 0
                             ? placeholder || selectValue || displayLabelKey
                             : onMore
-                            ? `${value?.length || '0'} selected`
-                            : `${value?.length || '0'} selected of ${
-                                allOptions.length
-                              }`
+                            ? format({
+                                id: 'selectMultiple.selectedMultipleNonTotal',
+                                messages,
+                                values: {
+                                  selectedCount: value?.length || 0,
+                                  totalCount: allOptions.length,
+                                },
+                              })
+                            : format({
+                                id: 'selectMultiple.selectedMultipleTotal',
+                                messages,
+                                values: {
+                                  selectedCount: value?.length || 0,
+                                  totalCount: allOptions.length,
+                                },
+                              })
```

Comment:
Can avoid the extra ternary (that lint is being told to ignore) and avoid duplication by deleting line 458 and replacing this with:
```suggestion
                            : format({
                                id: onMore
                                  ? 'selectMultiple.selectedMultipleNonTotal'
                                  : 'selectMultiple.selectedMultipleTotal',
                                messages,
                                values: {
                                  selectedCount: value?.length || 0,
                                  totalCount: allOptions.length,
                                },
                              })
```

Reply:
tanks for optimization